src/Callback.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Class Callback.
+ *
+ * Handles callbacks (also known as "notifications") from Ledyer.
+ */
+
+namespace Krokedil\Ledyer\Payments;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Callback.
+ */
+class Callback {
+
+	public const REST_API_NAMESPACE = 'krokedil/ledyer/payments/v1';
+	public const REST_API_ENDPOINT  = '/callback';
+	public const API_ENDPOINT       = 'wp-json/' . self::REST_API_NAMESPACE . self::REST_API_ENDPOINT;
+
+
+	/**
+	 * Callback constructor.
+	 */
+	public function __construct() {
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+		add_action( 'ledyer_payments_scheduled_callback', array( $this, 'handle_scheduled_callback' ) );
+	}
+
+	/**
+	 * Register the REST API route(s).
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		register_rest_route(
+			self::REST_API_NAMESPACE,
+			self::REST_API_ENDPOINT,
+			array(
+				'methods'             => 'POST',
+				'callback'            => array( $this, 'callback_handler' ),
+				'permission_callback' => '__return_true',
+			)
+		);
+	}
+
+	/**
+	 * Handles a callback ("notification") from Ledyer.
+	 *
+	 * @param \WP_REST_Request $request The REST request.
+	 * @return \WP_REST_Response|\WP_Error
+	 */
+	public function callback_handler( $request ) {
+		$context = array(
+			'filter'   => current_filter(),
+			'function' => __FUNCTION__,
+		);
+
+		$params = $request->get_json_params();
+		if ( empty( $params ) ) {
+			Ledyer_Payments()->logger()->debug( 'Received callback without parameters.', $context );
+			return new \WP_Error( 'missing_params', 'Missing parameters.', array( 'status' => 400 ) );
+		}
+
+		$params             = filter_var_array( $params, FILTER_SANITIZE_FULL_SPECIAL_CHARS );
```

Comment:
In this case, it should be safe if these conversions happen since the `sessionId` resembles a UUID, and the `eventType` is a reverse domain name notation.